### PR TITLE
fix: paste multiple lines in codeblock

### PR DESCRIPTION
**Pull request by MayurSMahajan**
Solves: #2992 

### Feature Preview

https://github.com/AppFlowy-IO/AppFlowy/assets/47064215/1afb675d-5e3f-4631-88d3-f5c0c27095d3

---

#### PR Checklist

- [x] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/code_block/code_block_shortcut_event.dart [hunk 1]
```diff
@@ -264,3 +278,30 @@ CommandShortcutEventHandler _selectAllInCodeBlockCommandHandler =
 
   return KeyEventResult.handled;
 };
+
+CommandShortcutEventHandler _pasteInCodeBlock = (editorState) {
+  final selection = editorState.selection;
+  if (selection == null || !selection.isCollapsed) {
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/code_block/code_block_shortcut_event.dart:284]: if selection isn't collapsed, we replace the selection with the pasted content, we shouldn't ignore the action, right?

**Comment by LucasXu0** [comment]
@MayurSMahajan Please include the tests.

**Comment by LucasXu0** [comment]
@MayurSMahajan I made some changes: only delete the selection in the code block paste command if all the selected nodes are code block, otherwise it will break the other paste commands.

**Comment by MayurSMahajan** [comment]
makes sense.

**Pull request merged**